tests/validation/tests/v3_api/test_group_grb.py

```diff
@@ -1,110 +1,386 @@
+def test_ggrb_1(remove_resource):
+    """ test that when a global role is assigned to a group,
+    all users in the group will get the permission;
+    when the ggrb is removed, all users in the group will lose
+    the permission.
 
-    r = requests.post(CATTLE_AUTH_URL, json={
-        'username': RANCHER_AUTH_USERNAME,
-        'password': RANCHER_AUTH_PASSWORD,
-        'responseType': 'json',
-    }, verify=False)
+    the default global role "catalogs-manage" is used
+    """
 
-    token = r.json()["token"]
+    target_group_name = "nestgroup1"
+    users = get_direct_user_by_group(target_group_name)
+    if NESTED_GROUP_ENABLED:
+        users = get_user_by_group(target_group_name)
+    # check that users can not create catalogs
+    for user in users:
+        validate_permission_create_catalog(user, False)
+    groups = search_group(target_group_name, ADMIN_TOKEN)
+    assert len(groups) == 1
+    ggrb = get_admin_client().create_global_role_binding(
+        globalRoleId="catalogs-manage", groupPrincipalId=groups[0]["id"])
+    # check that users can create catalogs now
+    for user in users:
+        catalog = validate_permission_create_catalog(user, True)
+        remove_resource(catalog)
+    # delete the ggrb
+    get_admin_client().delete(ggrb)
+    # check that users can not create catalogs
+    for user in get_user_by_group(target_group_name):
+        validate_permission_create_catalog(user, False)
 
-    testuser3_client = rancher.Client(
-        url=CATTLE_TEST_URL + "/v3",
-        token=token,
-        verify=False)
 
-    with pytest.raises(ApiError) as e:
-        rt = testuser3_client.create_role_template(name="rt-"+random_str())
+def test_ggrb_2(remove_resource):
+    """ test that when a global role is assigned to a group,
+    all users in the group get the permission from the role,
+    users not in the group do not get the permission
 
-    assert e.value.error.status == 403
-    assert e.value.error.code == 'Forbidden'
+    Steps:
+    - users in nestgroup2 cannot list clusters
+    - users not in nestgroup2 cannot list clusters
+    - create a custom global role gr1 that permits listing clusters
+    - create the global group role binding ggrb1 to bind gr1 to nestgroup2
+    - users in nestgroup2 can list clusters now
+    - users not in nestgroup2 still cannot list clusters
+    - delete the ggrb1
+    - users in nestgroup2 can not list clusters
+    """
 
-    gr = admin_client.create_global_role_binding(
-        globalRoleId="admin",
-        groupPrincipalId=groups[0]["id"])
+    target_group_name = "nestgroup2"
+    users = get_direct_user_by_group(target_group_name)
+    if NESTED_GROUP_ENABLED:
+        users = get_user_by_group(target_group_name)
+    # check that no users can not list clusters
+    for user in users:
+        validate_permission_list_cluster(user, 0)
+    # check that user nestedtestuser1 can not list clusters
+    validate_permission_list_cluster("nestedtestuser1", 0)
```

Comment:
Can we have  "nestedtestuser1" picked from the auth_info instead of hard coding it ?

Reply:
updated